mava/systems/sac/ff_isac.py

```diff
@@ -0,0 +1,600 @@
+from omegaconf import DictConfig, OmegaConf
+
+from mava.evaluator import make_eval_fns
+from mava.networks import FeedForwardActor as Actor
+from mava.networks import FeedForwardQNet as QNetwork
+from mava.systems.sac.types import (
+    BufferState,
+    LearnerState,
+    Metrics,
+    Networks,
+    Optimisers,
+    OptStates,
+    QVals,
+    QValsAndTarget,
+    SacParams,
+    Transition,
+)
+from mava.types import Observation
+from mava.utils import make_env as environments
+from mava.utils.checkpointing import Checkpointer
+from mava.utils.jax import unreplicate_batch_dim, unreplicate_n_dims
+from mava.utils.logger import LogEvent, MavaLogger
+from mava.utils.total_timestep_checker import check_total_timesteps
+from mava.wrappers import episode_metrics
+
+
+def init(
+    cfg: DictConfig,
+) -> Tuple[
+    Tuple[Environment, Environment],
+    Networks,
+    Optimisers,
+    TrajectoryBuffer,
+    LearnerState,
+    Array,
+    MavaLogger,
+    chex.PRNGKey,
+]:
+    """Initialize system by creating the envs, networks etc.
+
+    Args:
+        cfg: System configuration.
+
+    Returns:
+        Tuple containing:
+            Tuple[Environment, Environment]: The environment and evaluation environment.
+            Networks: Tuple of actor and critic networks.
+            Optimisers: Tuple of actor, critic and alpha optimisers.
+            TrajectoryBuffer: The replay buffer.
+            LearnerState: The initial learner state.
+            Array: The target entropy.
+            MavaLogger: The logger.
+            PRNGKey: The random key.
+    """
+    logger = MavaLogger(cfg)
+
+    key = jax.random.PRNGKey(cfg.system.seed)
+    devices = jax.devices()
+
+    def replicate(x: Any) -> Any:
+        """First replicate the update batch dim then put on devices."""
+        x = jax.tree_map(lambda y: jnp.broadcast_to(y, (cfg.system.update_batch_size, *y.shape)), x)
+        return jax.device_put_replicated(x, devices)
+
+    env, eval_env = environments.make(cfg)
+
+    n_agents = env.action_spec().shape[0]
+    action_dim = env.action_spec().shape[1]
+
+    key, actor_key, q1_key, q2_key, q1_target_key, q2_target_key = jax.random.split(key, 6)
+
+    acts = env.action_spec().generate_value()  # all agents actions
+    act_single_batched = acts[0][jnp.newaxis, ...]  # batch single agent action
+    obs = env.observation_spec().generate_value()
+    obs_single_batched = jax.tree_map(lambda x: x[0][jnp.newaxis, ...], obs)
+
+    # Making actor network
+    actor_torso = hydra.utils.instantiate(cfg.network.actor_network.pre_torso)
+    actor_action_head = hydra.utils.instantiate(
+        cfg.network.action_head, action_dim=env.action_dim, independent_std=False
+    )
+    actor = Actor(actor_torso, actor_action_head)
+    actor_params = actor.init(actor_key, obs_single_batched)
+
+    # Making Q networks
+    critic_torso = hydra.utils.instantiate(cfg.network.critic_network.pre_torso)
+    q = QNetwork(critic_torso)
```

Comment:
I just goes a bit more with the standard in the PPO files where we have `actor_network` and `critic_network` instead of just `actor` and `critic`. 
```suggestion
    q_network = QNetwork(critic_torso)
```